### PR TITLE
support RegExp input of distributed_files list

### PR DESCRIPTION
**Pull request by YuechengLiu**

### CODE REVIEW

File: parl/remote/client.py [hunk 1]
```diff
@@ -96,6 +99,15 @@ def read_local_files(self, distributed_files=[]):
...
         code_files = filter(lambda x: x.endswith('.py'), os.listdir('./'))
+        
+        parsed_distributed_files = []
+        for distributed_file in distributed_files:
+            parsed_list = glob.glob(distributed_file)
+            # exclude the directiories
+            for pathname in parsed_list:
+                if not os.path.isdir(pathname):
+                    parsed_distributed_files.append(pathname)
+        parsed_distributed_files = list(set(parsed_distributed_files))
```
Comment (TomorrowIsAnOtherDay) [inline parl/remote/client.py]: Why don't you declare `parsed_distributed_files` as a set?

**Review by TomorrowIsAnOtherDay — CHANGES_REQUESTED** [review]

**Comment by TomorrowIsAnOtherDay** [comment]
Please add the unit test and correct the code format according to the following page:
https://github.com/PaddlePaddle/PARL/blob/develop/docs/zh_CN/Overview.md#%E4%BB%A3%E7%A0%81%E8%B4%A1%E7%8C%AE%E8%A7%84%E8%8C%83

**Comment by liuyuecheng-github** [comment]
yes,sir.


| |
liuy150617
|
|
邮箱：liuy150617@163.com
|

Signature is customized by Netease Mail Master

On 08/11/2020 23:05, Bo Zhou wrote:

Please add the unit test and correct the code format according to the following page:
https://github.com/PaddlePaddle/PARL/blob/develop/docs/zh_CN/Overview.md#%E4%BB%A3%E7%A0%81%E8%B4%A1%E7%8C%AE%E8%A7%84%E8%8C%83

File: parl/remote/client.py [hunk 2]
```diff
@@ -96,6 +99,15 @@ def read_local_files(self, distributed_files=[]):
         pyfiles['other_files'] = {}
 
         code_files = filter(lambda x: x.endswith('.py'), os.listdir('./'))
+        
+        parsed_distributed_files = []
+        for distributed_file in distributed_files:
+            parsed_list = glob.glob(distributed_file)
```
Comment (zenghsh3) [inline parl/remote/client.py:100]: if the return list of glob.glob is empty, we should raise an exception.

File: parl/remote/client.py (hunk 1 above)
Comment (YuechengLiu) [reply parl/remote/client.py]: modified

File: parl/remote/client.py (hunk 2 above)
Comment (YuechengLiu) [reply parl/remote/client.py:100]: fixed

File: parl/remote/tests/support_RegExp_test.py [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+from parl.utils import logger
+
+
+@parl.remote_class
+class Actor(object):
+    def file_exists(self, filename):
+        return os.path.exists(filename)
+
+
+class TestCluster(unittest.TestCase):
```
Comment (zenghsh3) [inline parl/remote/tests/support_RegExp_test.py:34]: add a unittest where regex match failed.

Comment (YuechengLiu) [reply parl/remote/tests/support_RegExp_test.py:34]: added

**Review by zenghsh3 — DISMISSED** [review]
LGTM

File: parl/remote/tests/support_RegExp_test.py [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+        parl.connect(
+            'localhost:8602',
+            distributed_files=[
+                'distribute_test_dir/test*', 'distribute_test_dir/*npy'
+            ])
+        actor = Actor()
+        self.assertTrue(actor.file_exists('distribute_test_dir/test1.txt'))
+        self.assertTrue(actor.file_exists('distribute_test_dir/test2.txt'))
+        self.assertTrue(actor.file_exists('distribute_test_dir/data1.npy'))
+        self.assertTrue(actor.file_exists('distribute_test_dir/data2.npy'))
```
Comment (TomorrowIsAnOtherDay) [inline parl/remote/tests/support_RegExp_test.py:66]: add a test here:
`distribute_test_dir/data2.npy` does not exist.

Comment (YuechengLiu) [reply parl/remote/tests/support_RegExp_test.py:66]: added

**Review by TomorrowIsAnOtherDay — APPROVED** [review]

**Pull request merged**